driver/src/gnosis_safe_gas_station.rs

```diff
@@ -0,0 +1,107 @@
+//! Retrieves gas prices from the https://safe-relay.gnosis.io/ api call `/v1/gas-station`.
+
+use anyhow::Result;
+use ethcontract::U256;
+use isahc::http::uri::Uri;
+use isahc::prelude::*;
+use serde::Deserialize;
+use std::time::Duration;
+use uint::FromDecStrErr;
+
+/// The default uri at which the gas station api is available under.
+pub const DEFAULT_URI: &str = "https://safe-relay.gnosis.io/api/v1/gas-station/";
+
+/// Result of the api call. Prices are in wei.
+#[derive(Deserialize, Debug, Eq, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct GasPrice {
+    pub last_update: String,
+    #[serde(deserialize_with = "deserialize_u256_from_string")]
+    pub lowest: U256,
+    #[serde(deserialize_with = "deserialize_u256_from_string")]
+    pub safe_low: U256,
+    #[serde(deserialize_with = "deserialize_u256_from_string")]
+    pub standard: U256,
+    #[serde(deserialize_with = "deserialize_u256_from_string")]
+    pub fast: U256,
+    #[serde(deserialize_with = "deserialize_u256_from_string")]
+    pub fastest: U256,
+}
+
+#[cfg_attr(test, mockall::automock)]
+pub trait GetGasPrice {
+    /// Retrieves gas prices from the Gnosis Safe Relay api.
```

Comment:
This is the implementation detail of GasStation not specific to the GetGasPrice trait, which cold estimate the price based on any oracle it pleases (nit: I'd maybe name GasStation GnosisGasStation).

I personally also like -ing naming for interfaces (e.g. GasPriceEstimating) but that doesn't seem to be rust standard.

Reply:
In my mind this is the gnosis safe gas station, not a general gas station as seen by the return struct that is specific to Gnosis.  The reason I made it a trait is so that it is mockable. I don't think we need to generalize it for potential other gas stations yet.

Reply:
> I personally also like -ing naming for interfaces (e.g. GasPriceEstimating) but that doesn't seem to be rust standard.

Rust standard is to use infinitive verb forms. So `Read` and `Write` and `Display`, here I would say `EstimateGasPrice` by Rust standard. That being said, I like the -ing standard we use for traits in our codebase and I would recommend that here.